extras/two-factor-authentication.rst

```diff
@@ -0,0 +1,189 @@
+﻿Two-Factor Authentication
+=========================
+
+Two-factor authentication (2FA) enhances the security of your Zammad account by
+adding an extra layer of verification beyond a password. It requires you to
+provide two different types of authentication factors, typically something you
+know (like a password) and something you possess (like a mobile device or a
+security token), to ensure that you are an authorized individual that can access
+the account.
+
+.. hint::
+   Two-Factor Authentication is an optional feature. Administrators can learn
+   more :admin-docs:`here <settings/security/two-factor.html>`.
+
+Set Up Two-Factor Authentication
+--------------------------------
+
+If the system admin has enabled this feature, you can head to "Avatar -> Profile
+-> Password & Authentication" to set it up. Depending on the enabled two-factor
+methods, you may see one or more options in the table.
+
+To set up a two-factor method, use the ⋮ **Actions** menu next to it and choose
+**Set Up**.
+
+.. figure:: /images/extras/two-factor-authentication/user-profile-set-up.png
+   :alt: Set Up Two-Factor Method in Password & Authentication
+   :align: center
+
+In a modal dialog, you will be asked to confirm your current password.
+
+.. figure:: /images/extras/two-factor-authentication/password-check.png
+   :alt: Password Check Modal Dialog
+   :align: center
+
+.. _setup guide:
+
+Depending on the chosen two-factor method, you will be guided through the setup
+process, which includes specific steps.
+
+.. toctree::
+   :maxdepth: 1
+
+   two-factor-authentication/security-keys-setup
+   two-factor-authentication/authenticator-app-setup
+
+Sign-in with a Two-Factor Method
+--------------------------------
+
+When you set up two-factor authentication for your Zammad account, during the
+next sign-in you will be asked to provide the same two-factor method after
+entering correct username and password.
+
+Depending on the chosen two-factor method, this may be a security code, hardware
+key, etc.
+
+.. toctree::
+   :maxdepth: 1
+
+   two-factor-authentication/security-keys-sign-in
+   two-factor-authentication/authenticator-app-sign-in
+
+Trying Another Method
+~~~~~~~~~~~~~~~~~~~~~
+
+In case you are having issued during sign-in with your preferred two-factor
+authentication method, you can switch to another one, provided you have set it
+up previously.
```

Comment:
Maybe we should mention auto-generated recovery codes here? Even if one method was setup, user may have those anyway.

Reply:
Good point! I added more info about the recovery codes (and the fact that they are auto-generated) a few paragraphs below.